tensorflow_blade/build.py

```diff
@@ -120,23 +120,15 @@ def check_init_file_miss(path, ignore_root=False):
             raise Exception("missing __init__.py under " + dir)
 
 
-def check(args, check_py=False, check_cc=False):
-    if check_py:
-        with cwd(ROOT):
-            # every folder under python should contain a __init__.py file
-            check_init_file_miss("tf_blade", ignore_root=True)
-            # check tf_blade  dir
-            execute("{}/bin/black --check --diff tf_blade".format(args.python_dir))
-            execute("{}/bin/flake8 tf_blade".format(args.python_dir))
-            execute("{}/bin/mypy tf_blade".format(args.python_dir))
-            # check tests dir
-            check_init_file_miss("tests", ignore_root=True)
-            execute("{}/bin/black --check --diff tests".format(args.python_dir))
-            execute("{}/bin/flake8 tests".format(args.python_dir))
-            execute("{}/bin/mypy tests".format(args.python_dir))
-    if check_cc:
-        # TODO(xiafei.qiuxf): add cxx check.
-        pass
+# No need to do cc check, since pre-commit do cc check with clang-format
+def check(args):
```

Comment:
Nit: Our `build.py` currently does a lot. of things. It does configurations, style checking, bazel building, packaging, and testing. I think we should deprecate it.

1. Install python3 dependencies by specifying requirements and deprecate venv
2. Configure bazel build with user bazelrc
3. Move these style checks into pre-commit scripts. 
4.  I prefer to use `setuptools` to build package wheel. Also, rename `build.py` to `setup.py`.
5. Call bazel building with custom extension in `setup.py`
6. Also call python unit test with `python -m unittest discover tests/....`(I don't think use bazel cache python3 UT is a good option before we can write dependencies correctly all the time.

Reply:
1. Now still using venv since we want to share the same build.py for internal version build.
2. Only .bazelrc is not enough for different build settings.
3. Doing check in pre-commit for cc is simple, but py check includes falke8/isort/mypy check, I don't konw whether pre-commit is capable of doing this.
4. We have a setup.py to build package wheel, it's not commited yet.
About 5 && 6, I still want follow the way we did in internal version, i.e. calling bazel directly. As regard to bazel cache, I have no preference now.

Reply:
As for 3: pre-commit can do flake8/isort check, however I think mypy check should still be here, since mypy check errors cannot be automatically fixed.